transport/stream.go

```diff
@@ -1,10 +1,40 @@
 package transport
 
 import (
+	"fmt"
+	"math"
+	"math/bits"
 	"scylla-go-driver/frame"
 )
 
 type StreamIDAllocator interface {
 	Alloc() (frame.StreamID, error)
 	Free(id frame.StreamID)
 }
+
+const maxStreamID = math.MaxInt16
+
+const bucketSize = 64
+const buckets = (maxStreamID + 1) / bucketSize
+
+// DefaultStreamIDAllocator is a StreamIDAllocator that always allocates the smallest possible stream on Alloc().
+type DefaultStreamIDAllocator struct {
+	usedBitmap [buckets]uint64
+}
+
+func (s *DefaultStreamIDAllocator) Alloc() (frame.StreamID, error) {
```

Comment:
Since this only holds a map you can pass it by value always.

Reply:
Correct me if I'm wrong but i think that the pointer receiver is needed as both Alloc() and Free() need to modify the map.

Reply:
A) this is not a map - please keep the pointers - my bad.
B) map itself is kept on heap and what you have as map[X]Y is a pointer already, this is not like in c++